protos/flyteidl/service/dataproxy.proto

```diff
@@ -44,6 +44,25 @@ message CreateUploadLocationRequest {
   bytes content_md5 = 5;
 }
 
+// CreateDownloadLocationRequest specified request for the CreateDownloadLocation API.
+message CreateDownloadLocationRequest {
+  // NativeUrl specifies the url in the format of the configured storage provider (e.g. s3://my-bucket/randomstring/suffix.tar)
+  string native_url = 1;
+
+  // ExpiresIn defines a requested expiration duration for the generated url. The request will be rejected if this
+  // exceeds the platform allowed max.
+  // +optional. The default value comes from a global config.
+  google.protobuf.Duration expires_in = 2;
+
+}
+
+message CreateDownloadLocationResponse {
```

Comment:
Should this also not say permission denied or is that an error

Reply:
If an error should it not be explicitly called

Reply:
Errors will return grpc errors.. there are built-in ones for Access Denied